packages/grid/_modules_/grid/hooks/features/useComponents.tsx

```diff
@@ -33,7 +34,8 @@ const wrapWithProps: <TProps, StaticProps>(
 };
 
 export const useComponents = (
-  componentsProp: GridSlotsComponent | undefined,
+  componentSlotsProp: GridSlotsComponent | undefined,
+  componentsProp: GridSlotsComponentProps | undefined,
```

Comment:
How about we avoid renaming the variables? Maybe something consistent (`Prop` suffix and `GridSlots` prefix) like this?

```suggestion
  componentsProp: GridSlotsComponents | undefined,
  componentsPropsProp: GridSlotsComponentsProps | undefined,
```

Reply:
This is not part of the public api. It's just internal to the hook to differentiate the 2 notions.
PropsProp feels a bit weird

Reply:
I was trying to optimize for making it easier for somebody new to this part of the codebase to understand what's going on.

Reply:
In this case, could we keep it semantically correct:

```suggestion
  componentSlotsProp: GridSlotsComponent | undefined,
  componentsProps: GridSlotsComponentProps | undefined,
```